--- a/include/ssp/string.h
+++ b/include/ssp/string.h
@@ -36,7 +36,7 @@

 __BEGIN_DECLS
 void *__memcpy_chk(void *, const void *, size_t, size_t);
-void *__memmove_chk(void *, void *, size_t, size_t);
+void *__memmove_chk(void *, const void *, size_t, size_t);
 void *__mempcpy_chk(void *, const void *, size_t, size_t);
 void *__memset_chk(void *, int, size_t, size_t);
 char *__stpcpy_chk(char *, const char *, size_t);

--- a/include/ssp/stdlib.h
+++ b/include/ssp/stdlib.h
@@ -20,7 +20,7 @@
 __ssp_decl(int, wctomb, (char *__buf, wchar_t __wc))
 {
   if (__buf != NULL)
-    __ssp_check(__buf, MB_CUR_MAX, __ssp_bos);
+    __ssp_check(__buf, (unsigned int) MB_CUR_MAX, __ssp_bos);
   return __ssp_real_wctomb (__buf, __wc);
 }